src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -169,10 +184,32 @@ private static long[] getBlockStates(CompoundTag nbt, String subReg) {
      * @return if the current block is part of the subregion.
      */
     private static boolean inSubregion(CompoundTag nbt, String subReg, int x, int y, int z) {
-        return x >= 0 && y >= 0 && z >= 0 &&
-                x < Math.abs(nbt.getCompound("Regions").getCompound(subReg).getCompound("Size").getInt("x")) &&
-                y < Math.abs(nbt.getCompound("Regions").getCompound(subReg).getCompound("Size").getInt("y")) &&
-                z < Math.abs(nbt.getCompound("Regions").getCompound(subReg).getCompound("Size").getInt("z"));
+
+        CompoundTag region = nbt.getCompound("Regions").getCompound(subReg);
+
+        int xSize = region.getCompound("Size").getInt("x");
+        int ySize = region.getCompound("Size").getInt("y");
+        int zSize = region.getCompound("Size").getInt("z");
+
+        int xPos = region.getCompound("Position").getInt("x");
+        int yPos = region.getCompound("Position").getInt("y");
+        int zPos = region.getCompound("Position").getInt("z");
+
+        int xMin = (xSize >= 0) ? xPos : xPos + xSize + 1;
+        int yMin = (ySize >= 0) ? yPos : yPos + ySize + 1;
+        int zMin = (zSize >= 0) ? zPos : zPos + zSize + 1;
+
+        int xMax = xMin + Math.abs(xSize);
+        int yMax = yMin + Math.abs(ySize);
+        int zMax = zMin + Math.abs(zSize);
+
+        boolean withinX = xSize != 0 && x >= xMin && x < xMax;
+        boolean withinY = ySize != 0 && y >= yMin && y < yMax;
+        boolean withinZ = zSize != 0 && z >= zMin && z < zMax;
+
+        boolean found = withinX && withinY && withinZ;
+
+        return found;
```

Comment:
a bit lengthy but not wrong as far as i can tell.
the xSize != 0 check is redundant (if size is 0, xMax = xMin  and x cant be >= and < to the same number)